Automattic/wp-calypso
# Enable coming soon v2 for /start site functionality

This is a break out from #45896 props to @p-jackson for nudging me in the right direction here.

#### Changes proposed in this Pull Request

* Enable creation of sites under coming soon v2 mode
* Replaces and drops `getNewSitePublicSetting`, `getNewSiteComingSoonSetting`, and `shouldNewSiteBePrivateByDefault` constant functions with explicitly labeled Visibility enum usage.


#### Testing instructions

* https://calypso.localhost:3000/start?flags=coming-soon-v2

When the flag is enabled coming soon v2 should be enabled.

Site options should match:
```
blog_public = 0
wpcom_public_coming_soon = 1
wpcom_coming_soon = 0
```

* https://calypso.localhost:3000/start

When flag is disabled, coming soon v1 should be enabled.

Site options should match existing behavior:
```
blog_public = 0
wpcom_coming_soon = 1
```
With the addition of the v2 option being disabled:
```
wpcom_public_coming_soon = 0
```
